corehq/messaging/scheduling/static/scheduling/js/conditional_alert_list.js

```diff
@@ -120,8 +120,13 @@ hqDefine("scheduling/js/conditional_alert_list", [
                 },
                 success: function (data) {
                     self.showPaginationSpinner(false);
+                    if (_.isString(data)) {
+                        // stop polling if data is HTML (login or error page)
+                        return;
```

Comment:
What would you think of calling `reloadPageSoon` in either case, but giving it a longer delay (like 10 min or 30 min) in this case? I don't think it's a clear-cut better approach, just another idea.

Reply:
I wouldn't have a problem with that, but it would complicate the implementation a bit, and I don't feel like we have time right now to polish this cannonball.

I don't  like how this is implemented because it's fairly intricate and has to be done correctly in each place where a page uses polling. One potential better solution would be to implement a generic polling helper/pattern that can be used everywhere and that does all the right things at the right times. Another (probably better solution) is to try to fix the issue in the 2FA code so a polling page does not disrupt the login process. I think probably both of those things should be done eventually.